Dependency upgrades

In build.gradle

google_api_protos: 'com.google.api.grpc:proto-google-common-protos:1.0.0' --> 1.12.0
com.google.auth:google-auth-library-credentials:0.9.0 to 0.9.1
com.google.code.findbugs:jsr305:3.0.0 to 3.0.2
protobufVersion 3.5.1 --> 3.6.0

There may be others but these are the ones that are conflicting upstream (that is, other versions already being selected) in google-cloud-java

@ejona86 Is this the breakage you're referring to? I just saw this at head. If so,  I guess that's a separate issue to fix?

```
> Task :grpc-compiler:compileJava_pluginExecutableJava_pluginCpp
In file included from /Users/elharo/grpc-java/compiler/src/java_plugin/cpp/java_plugin.cpp:8:
/Users/elharo/grpc-java/compiler/src/java_plugin/cpp/java_generator.h:8:10: fatal error: 'google/protobuf/io/zero_copy_stream.h' file not found
#include <google/protobuf/io/zero_copy_stream.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

/Users/elharo/grpc-java/compiler/src/java_plugin/cpp/java_generator.h:8:10: fatal error: 'google/protobuf/io/zero_copy_stream.h' file not found
#include <google/protobuf/io/zero_copy_stream.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

In file included from /Users/elharo/grpc-java/compiler/src/java_plugin/cpp/java_generator.cpp:1:
/Users/elharo/grpc-java/compiler/src/java_plugin/cpp/java_generator.h:8:10: fatal error: 'google/protobuf/io/zero_copy_stream.h' file not found
#include <google/protobuf/io/zero_copy_stream.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/alts/build.gradle
+++ b/alts/build.gradle
@@ -27,10 +27,10 @@ dependencies {
             libraries.lang,
             libraries.protobuf
     compile (libraries.google_auth_oauth2_http) {
-        // prefer 3.0.2 from libraries instead of 1.3.9
-        exclude group: 'com.google.code.findbugs', module: 'jsr305'
-        // prefer 20.0 from libraries instead of 19.0
+        // prefer 26.0-android from libraries instead of 25.1-android
         exclude group: 'com.google.guava', module: 'guava'
+        // prefer 0.19.2 from libraries instead of 0.18.0
+        exclude group: 'io.opencensus', module: 'opencensus-api'
     }
     compileOnly libraries.javax_annotation
     runtime project(':grpc-grpclb')

--- a/build.gradle
+++ b/build.gradle
@@ -108,7 +108,7 @@ subprojects {
         javaPluginPath = "$rootDir/compiler/build/exe/java_plugin/$protocPluginBaseName$exeSuffix"
 
         nettyVersion = '4.1.32.Final'
-        googleauthVersion = '0.9.0'
+        googleauthVersion = '0.13.0'
         guavaVersion = '26.0-android'
         protobufVersion = '3.6.1'
         protocVersion = protobufVersion

